src/nets/e2e_asr_th.py

```diff
@@ -66,20 +66,25 @@ def pad_list(xs, pad_value):
     return pad
 
 
-def fill_padded_part(xs, ilens, fill_value):
-    """Fucntion to fill padded part with selected value
+def make_pad_mask(lengths):
+    """Function to make mask tensor containing indices of padded part
 
-    :param torch.Tensor xs: tensor (B, Tmax, ...)
-    :param torch.Tensor ilens:  list of lengths (B)
-    :param float fill_value:  value to fill padded part
-    :return: xs whose padded parts are filled by fill_value
+    e.g.: lengths = [5, 3, 2]
+          mask = [[0, 0, 0, 0 ,0],
+                  [0, 0, 0, 1, 1],
+                  [0, 0, 1, 1, 1]]
+
+    :param list lengths: list of lengths (B)
+    :return: mask tensor contraining indices of padded part (B, Tmax)
+    :rtype: torch.Tensor
     """
-    assert xs.size(0) == len(ilens)
-    new_xs = xs.new(*xs.size()).fill_(fill_value)
-    for idx, l in enumerate(ilens):
-        new_xs[idx, :l] = xs[idx, :l]
+    bs = int(len(lengths))
+    maxlen = int(max(lengths))
+    mask = torch.zeros(bs, maxlen).byte()
```

Comment:
use torch.zeros(bs, maxlen, dtype=torch.uint8) because it can byte-allocate once instead of twice as float-allocate ->byte-allocate